File: PYTHON/Program/Penjual.py
from User import User

class Penjual(User):
    def __init__(self, id_user, nama, email, password, no_hp,
                 id_toko, nama_toko, alamat_toko):
        super().__init__(id_user, nama, email, password, no_hp)
        self.id_toko = id_toko
        self.nama_toko = nama_toko
        self.alamat_toko = alamat_toko
        self.daftar_produk = []

    def tambah_produk(self, produk):
        self.daftar_produk.append(produk)

    def tampilkan_info(self):
        super().tampilkan_info()
        print(f"ID Toko   : {self.id_toko}")
        print(f"Nama Toko : {self.nama_toko}")
        print(f"Alamat    : {self.alamat_toko}")
        print("=== Daftar Produk ===")
        for p in self.daftar_produk:
            p.tampilkan_info()
            print("----------------------")

File: JAVA/Program/Penjual.java
import java.util.ArrayList;

public class Penjual extends User {
    private String idToko, namaToko, alamatToko;
    private ArrayList<Produk> daftarProduk = new ArrayList<>();

    public Penjual(String idUser, String nama, String email, String password, String noHp,
                   String idToko, String namaToko, String alamatToko) {
        super(idUser, nama, email, password, noHp);
        this.idToko = idToko;
        this.namaToko = namaToko;
        this.alamatToko = alamatToko;
    }

    public void tambahProduk(Produk p) {
        daftarProduk.add(p);
    }

    @Override
    public void tampilkanInfo() {
        super.tampilkanInfo();
        System.out.println("ID Toko   : " + idToko);
        System.out.println("Nama Toko : " + namaToko);
        System.out.println("Alamat    : " + alamatToko);
        System.out.println("=== Daftar Produk ===");
        for (Produk p : daftarProduk) {
            p.tampilkanInfo();
            System.out.println("----------------------");
        }
    }
}
